instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/AlternativeRouteSelectionTest.kt

```diff
@@ -81,6 +86,7 @@ class AlternativeRouteSelectionTest : BaseTest<BasicNavigationViewActivity>(
         }
     }
 
+    @Ignore
```

Comment:
@cafesilencio Could this https://github.com/mapbox/mapbox-navigation-android/pull/4089 be related?

Reply:
It could be. I revert that change on a branch and see what the result is.

Reply:
It also happens with SanityUIRouteTest.

Reply:
It fails at BasicNavigationViewActivity -> onCreate -> binding = ActivityBasicNavigationViewBinding...

I think this is a bigger issue outside the scope of this PR.  I tried a quick and dirty fix to remove the binding related calls and set the content view the old fashioned way using the technique we were doing earlier in our test activities where the view layout had a layout container and we were adding the mapview during onCreate and this was also failing. Further investigation is needed.

Reply:
I'm able to run the test and have no build issues, but it's failing:

```
java.lang.AssertionError: expected:<DirectionsRoute{routeIndex=1, distance=1872.7, duration=261.6, durationTypical=null, geometry=mvl`gAnalshFa@iVUuMG}D[{Rq@y`@ScLaB{aA{@gi@[kQGmEIqE[yQ{@uh@aAgl@_@{Ta@qU_Agk@aB{aA{@_j@UmNKoGMoGS{Mm@i^GyEGqDysBhGkg@~AaZ|@kPh@}AD{[bAonB`GeDJmDJkzArEqRl@{FPR|K\jT`@vVJ|GTlN, weight=387.3, weightName=routability, legs=[RouteLeg{distance=1872.7, duration=261.6, durationTypical=null, summary=Fulton Street, 24th Avenue, admins=null, steps=[LegStep{distance=911.0, duration=99.4, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=mvl`gAnalshFa@iVUuMG}D[{Rq@y`@ScLaB{aA{@gi@[kQGmEIqE[yQ{@uh@aAgl@_@{Ta@qU_Agk@aB{aA{@_j@UmNKoGMoGS{Mm@i^GyEGqD, name=Fulton Street, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.49348, 37.772151], bearingBefore=0.0, bearingAfter=87.0, instruction=Head east on Fulton Street, type=depart, modifier=null, exit=null}, voiceInstructions=[VoiceInstructions{distanceAlongGeometry=911.0, announcement=Head east on Fulton Street, then turn left onto 24th Avenue, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Head east on Fulton Street, then turn left onto <say-as interpret-as="address">24th</say-as> Avenue</prosody></amazon:effect></speak>}, VoiceInstructions{distanceAlongGeometry=137.5, announcement=Turn left onto 24th Avenue, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Turn left onto <say-as interpret-as="address">24th</say-as> Avenue</prosody></amazon:effect></speak>}], bannerInstructions=[BannerInstructions{distanceAlongGeometry=911.0, primary=BannerText{text=24th Avenue, components=[BannerComponents{text=24th Avenue, type=text, subType=null, abbreviation=24th Ave, abbreviationPriority=0, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=turn, modifier=left, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}], drivingSide=right, weight=156.2, intersections=[StepIntersection{rawLocation=[-122.49348, 37.772151], bearings=[87], classes=null, entry=[true], in=null, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.492777, 37.772183], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.491708, 37.772232], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.490638, 37.772281], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.489565, 37.772329], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, true, false], in=3, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.488492, 37.772378], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.487418, 37.772427], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.486349, 37.772476], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.485279, 37.772525], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.484208, 37.772572], bearings=[0, 90, 150, 210, 270], classes=null, entry=[false, true, false, true, false], in=4, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}, LegStep{distance=831.5, duration=129.7, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=wsm`gA|zwrhFysBhGkg@~AaZ|@kPh@}AD{[bAonB`GeDJmDJkzArEqRl@, name=24th Avenue, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.483135, 37.77262], bearingBefore=85.0, bearingAfter=355.0, instruction=Turn left onto 24th Avenue, type=turn, modifier=left, exit=null}, voiceInstructions=[VoiceInstructions{distanceAlongGeometry=811.5, announcement=Continue on 24th Avenue for a half mile, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Continue on <say-as interpret-as="address">24th</say-as> Avenue for a half mile</prosody></amazon:effect></speak>}, VoiceInstructions{distanceAlongGeometry=448.8, announcement=In a quarter mile, turn left onto Geary Boulevard, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">In a quarter mile, turn left onto Geary Boulevard</prosody></amazon:effect></speak>}, VoiceInstructions{distanceAlongGeometry=96.2, announcement=Turn left onto Geary Boulevard, then you will arrive at your destination, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Turn left onto Geary Boulevard, then you will arrive at your destination</prosody></amazon:effect></speak>}], bannerInstructions=[BannerInstructions{distanceAlongGeometry=831.5, primary=BannerText{text=Geary Boulevard, components=[BannerComponents{text=Geary Boulevard, type=text, subType=null, abbreviation=Geary Blvd, abbreviationPriority=0, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=turn, modifier=left, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}], drivingSide=right, weight=141.9, intersections=[StepIntersection{rawLocation=[-122.483135, 37.77262], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.483268, 37.774489], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, false, true], in=2, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.483405, 37.776355], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, false, true], in=2, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.48354, 37.778222], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, false, true], in=2, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}, LegStep{distance=130.2, duration=32.5, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=gf|`gAt|xrhF{FPR|K\jT`@vVJ|GTlN, name=Geary Boulevard, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.483675, 37.780084], bearingBefore=355.0, bearingAfter=355.0, instruction=Turn left onto Geary Boulevard, type=turn, modifier=left, exit=null}, voiceInstructions=[VoiceInstructions{distanceAlongGeometry=20.0, announcement=You have arrived at your destination, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">You have arrived at your destination</prosody></amazon:effect></speak>}], bannerInstructions=[BannerInstructions{distanceAlongGeometry=130.2, primary=BannerText{text=You will arrive, components=[BannerComponents{text=You will arrive, type=text, subType=null, abbreviation=null, abbreviationPriority=null, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=arrive, modifier=straight, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}, BannerInstructions{distanceAlongGeometry=20.0, primary=BannerText{text=You have arrived, components=[BannerComponents{text=You have arrived, type=text, subType=null, abbreviation=null, abbreviationPriority=null, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=arrive, modifier=straight, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}], drivingSide=right, weight=89.2, intersections=[StepIntersection{rawLocation=[-122.483675, 37.780084], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, false, false], in=2, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.483684, 37.78021], bearings=[0, 90, 180, 270], classes=null, entry=[true, false, false, true], in=2, out=3, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.484756, 37.780162], bearings=[0, 90, 180, 270], classes=null, entry=[true, false, false, true], in=1, out=3, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}, LegStep{distance=0.0, duration=0.0, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=mj|`gAto{rhF, name=Geary Boulevard, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.485003, 37.780151], bearingBefore=267.0, bearingAfter=0.0, instruction=You have arrived at your destination, type=arrive, modifier=null, exit=null}, voiceInstructions=[], bannerInstructions=[], drivingSide=right, weight=0.0, intersections=[StepIntersection{rawLocation=[-122.485003, 37.780151], bearings=[87], classes=null, entry=[true], in=0, out=null, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}], incidents=null, annotation=LegAnnotation{distance=null, duration=null, speed=null, maxspeed=[MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=40, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=40, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=40, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=40, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=40, unit=km/h, unknown=null, none=null}], congestion=null}}], routeOptions=RouteOptions{baseUrl=https://api.mapbox.com, user=mapbox, profile=driving-traffic, coordinates=[Point{type=Point, bbox=null, coordinates=[-122.4934801, 37.7721532]}, Point{type=Point, bbox=null, coordinates=[-122.4850055, 37.7801765]}], alternatives=true, language=en, radiuses=null, bearings=null, continueStraight=false, roundaboutExits=false, geometries=polyline6, overview=full, steps=true, annotations=maxspeed, exclude=null, voiceInstructions=true, bannerInstructions=true, voiceUnits=imperial, accessToken=tokenOmitted, requestUuid=ckl8j5te2ch2i79r7cmgw67sj, approaches=null, waypointIndices=null, waypointNames=null, waypointTargets=null, walkingOptions=null}, voiceLanguage=en-US}> but was:<DirectionsRoute{routeIndex=0, distance=1683.8, duration=207.3, durationTypical=null, geometry=mvl`gAnalshFa@iVUuMG}D[{Rq@y`@ScLaB{aA{@gi@[kQGmEIqE[yQ{@uh@aAgl@_@{Ta@qU_Agk@wl@hBiRj@eELag@|AiDJkELimB~FusBlGwAg|@IuDGwDoAwu@GmEsDLcIVooAxDqCHoFPaGP}FPTlN, weight=365.4, weightName=routability, legs=[RouteLeg{distance=1683.8, duration=207.3, durationTypical=null, summary=Fulton Street, 27th Avenue, admins=null, steps=[LegStep{distance=628.0, duration=63.099999999999994, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=mvl`gAnalshFa@iVUuMG}D[{Rq@y`@ScLaB{aA{@gi@[kQGmEIqE[yQ{@uh@aAgl@_@{Ta@qU_Agk@, name=Fulton Street, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.49348, 37.772151], bearingBefore=0.0, bearingAfter=87.0, instruction=Head east on Fulton Street, type=depart, modifier=null, exit=null}, voiceInstructions=[VoiceInstructions{distanceAlongGeometry=628.0, announcement=Head east on Fulton Street, then turn left onto 27th Avenue, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Head east on Fulton Street, then turn left onto <say-as interpret-as="address">27th</say-as> Avenue</prosody></amazon:effect></speak>}, VoiceInstructions{distanceAlongGeometry=149.3, announcement=Turn left onto 27th Avenue, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Turn left onto <say-as interpret-as="address">27th</say-as> Avenue</prosody></amazon:effect></speak>}], bannerInstructions=[BannerInstructions{distanceAlongGeometry=628.0, primary=BannerText{text=27th Avenue, components=[BannerComponents{text=27th Avenue, type=text, subType=null, abbreviation=27th Ave, abbreviationPriority=0, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=turn, modifier=left, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}], drivingSide=right, weight=107.7, intersections=[StepIntersection{rawLocation=[-122.49348, 37.772151], bearings=[87], classes=null, entry=[true], in=null, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.492777, 37.772183], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.491708, 37.772232], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.490638, 37.772281], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.489565, 37.772329], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, true, false], in=3, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.488492, 37.772378], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.487418, 37.772427], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}, LegStep{distance=623.9, duration=72.3, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=wjm`gAxc~rhFwl@hBiRj@eELag@|AiDJkELimB~FusBlG, name=27th Avenue, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.486349, 37.772476], bearingBefore=85.0, bearingAfter=355.0, instruction=Turn left onto 27th Avenue, type=turn, modifier=left, exit=null}, voiceInstructions=[VoiceInstructions{distanceAlongGeometry=603.9, announcement=In a half mile, turn right onto Anza Street, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">In a half mile, turn right onto Anza Street</prosody></amazon:effect></speak>}, VoiceInstructions{distanceAlongGeometry=129.4, announcement=Turn right onto Anza Street, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Turn right onto Anza Street</prosody></amazon:effect></speak>}], bannerInstructions=[BannerInstructions{distanceAlongGeometry=623.9, primary=BannerText{text=Anza Street, components=[BannerComponents{text=Anza Street, type=text, subType=null, abbreviation=Anza St, abbreviationPriority=0, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=turn, modifier=right, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}], drivingSide=right, weight=84.9, intersections=[StepIntersection{rawLocation=[-122.486349, 37.772476], bearings=[0, 90, 270], classes=null, entry=[true, true, false], in=2, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.486484, 37.774342], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, false, true], in=2, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.486619, 37.776209], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, false, true], in=2, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}, LegStep{distance=188.6, duration=33.1, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=whx`gAb}~rhFwAg|@IuDGwDoAwu@GmE, name=Anza Street, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.486754, 37.778076], bearingBefore=355.0, bearingAfter=85.0, instruction=Turn right onto Anza Street, type=turn, modifier=right, exit=null}, voiceInstructions=[VoiceInstructions{distanceAlongGeometry=168.6, announcement=In 600 feet, turn left onto 25th Avenue, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">In 600 feet, turn left onto <say-as interpret-as="address">25th</say-as> Avenue</prosody></amazon:effect></speak>}, VoiceInstructions{distanceAlongGeometry=85.5, announcement=Turn left onto 25th Avenue, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Turn left onto <say-as interpret-as="address">25th</say-as> Avenue</prosody></amazon:effect></speak>}], bannerInstructions=[BannerInstructions{distanceAlongGeometry=188.6, primary=BannerText{text=25th Avenue, components=[BannerComponents{text=25th Avenue, type=text, subType=null, abbreviation=25th Ave, abbreviationPriority=0, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=turn, modifier=left, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}], drivingSide=right, weight=77.6, intersections=[StepIntersection{rawLocation=[-122.486754, 37.778076], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, false, true], in=2, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.485683, 37.778125], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, true, false], in=3, out=1, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}, LegStep{distance=207.4, duration=23.5, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=ynx`gAfwzrhFsDLcIVooAxDqCHoFPaGP, name=25th Avenue, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.484612, 37.778173], bearingBefore=85.0, bearingAfter=355.0, instruction=Turn left onto 25th Avenue, type=turn, modifier=left, exit=null}, voiceInstructions=[VoiceInstructions{distanceAlongGeometry=187.4, announcement=In 700 feet, turn left onto Geary Boulevard, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">In 700 feet, turn left onto Geary Boulevard</prosody></amazon:effect></speak>}, VoiceInstructions{distanceAlongGeometry=132.4, announcement=Turn left onto Geary Boulevard, then you will arrive at your destination, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Turn left onto Geary Boulevard, then you will arrive at your destination</prosody></amazon:effect></speak>}], bannerInstructions=[BannerInstructions{distanceAlongGeometry=207.4, primary=BannerText{text=Geary Boulevard, components=[BannerComponents{text=Geary Boulevard, type=text, subType=null, abbreviation=Geary Blvd, abbreviationPriority=0, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=turn, modifier=left, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}], drivingSide=right, weight=35.5, intersections=[StepIntersection{rawLocation=[-122.484612, 37.778173], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, true, false], in=3, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}, LegStep{distance=35.9, duration=15.3, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=ec|`gAt_{rhF}FPTlN, name=Geary Boulevard, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.484747, 37.780035], bearingBefore=355.0, bearingAfter=355.0, instruction=Turn left onto Geary Boulevard, type=turn, modifier=left, exit=null}, voiceInstructions=[VoiceInstructions{distanceAlongGeometry=11.7, announcement=You have arrived at your destination, ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">You have arrived at your destination</prosody></amazon:effect></speak>}], bannerInstructions=[BannerInstructions{distanceAlongGeometry=35.9, primary=BannerText{text=You will arrive, components=[BannerComponents{text=You will arrive, type=text, subType=null, abbreviation=null, abbreviationPriority=null, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=arrive, modifier=straight, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}, BannerInstructions{distanceAlongGeometry=11.7, primary=BannerText{text=You have arrived, components=[BannerComponents{text=You have arrived, type=text, subType=null, abbreviation=null, abbreviationPriority=null, imageBaseUrl=null, imageUrl=null, directions=null, active=null}], type=arrive, modifier=straight, degrees=null, drivingSide=null}, secondary=null, sub=null, view=null}], drivingSide=right, weight=59.7, intersections=[StepIntersection{rawLocation=[-122.484747, 37.780035], bearings=[0, 90, 180, 270], classes=null, entry=[true, true, false, false], in=2, out=0, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}, StepIntersection{rawLocation=[-122.484756, 37.780162], bearings=[0, 90, 180, 270], classes=null, entry=[true, false, false, true], in=2, out=3, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}, LegStep{distance=0.0, duration=0.0, durationTypical=null, speedLimitUnit=mph, speedLimitSign=mutcd, geometry=mj|`gAto{rhF, name=Geary Boulevard, ref=null, destinations=null, mode=driving, pronunciation=null, rotaryName=null, rotaryPronunciation=null, maneuver=StepManeuver{rawLocation=[-122.485003, 37.780151], bearingBefore=267.0, bearingAfter=0.0, instruction=You have arrived at your destination, type=arrive, modifier=null, exit=null}, voiceInstructions=[], bannerInstructions=[], drivingSide=right, weight=0.0, intersections=[StepIntersection{rawLocation=[-122.485003, 37.780151], bearings=[87], classes=null, entry=[true], in=0, out=null, lanes=null, geometryIndex=null, isUrban=null, adminIndex=null, restStop=null, tollCollection=null, mapboxStreetsV8=null, tunnelName=null}], exits=null}], incidents=null, annotation=LegAnnotation{distance=null, duration=null, speed=null, maxspeed=[MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=56, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=40, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=40, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=null, unit=null, unknown=true, none=null}, MaxSpeed{speed=40, unit=km/h, unknown=null, none=null}, MaxSpeed{speed=40, unit=km/h, unknown=null, none=null}], congestion=null}}], routeOptions=RouteOptions{baseUrl=https://api.mapbox.com, user=mapbox, profile=driving-traffic, coordinates=[Point{type=Point, bbox=null, coordinates=[-122.4934801, 37.7721532]}, Point{type=Point, bbox=null, coordinates=[-122.4850055, 37.7801765]}], alternatives=true, language=en, radiuses=null, bearings=null, continueStraight=false, roundaboutExits=false, geometries=polyline6, overview=full, steps=true, annotations=maxspeed, exclude=null, voiceInstructions=true, bannerInstructions=true, voiceUnits=imperial, accessToken=tokenOmitted, requestUuid=ckl8j5te2ch2i79r7cmgw67sj, approaches=null, waypointIndices=null, waypointNames=null, waypointTargets=null, walkingOptions=null}, voiceLanguage=en-US}>
at org.junit.Assert.fail(Assert.java:88)
at org.junit.Assert.failNotEquals(Assert.java:834)
at org.junit.Assert.assertEquals(Assert.java:118)
at org.junit.Assert.assertEquals(Assert.java:144)
at com.mapbox.navigation.instrumentation_tests.ui.routeline.AlternativeRouteSelectionTest$selectAlternateRoute$1$2.onRouteProgressChanged(AlternativeRouteSelectionTest.kt:116)
at com.mapbox.navigation.core.trip.session.MapboxTripSession.updateRouteProgress(MapboxTripSession.kt:537)
at com.mapbox.navigation.core.trip.session.MapboxTripSession.access$updateRouteProgress(MapboxTripSession.kt:50)
at com.mapbox.navigation.core.trip.session.MapboxTripSession$updateDataFromNavigatorStatus$updateNavigatorStatusDataJob$1.invokeSuspend(MapboxTripSession.kt:507)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at androidx.test.espresso.base.Interrogator.loopAndInterrogate(Interrogator.java:148)
at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:519)
at androidx.test.espresso.base.UiControllerImpl.loopMainThreadUntilIdle(UiControllerImpl.java:441)
at androidx.test.espresso.Espresso$1.run(Espresso.java:277)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:223)
at android.app.ActivityThread.main(ActivityThread.java:7656)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

Reply:
I know, SanityUIRouteTest also fails.  It's not a build problem there is some other issue with the instrumentation tests outside of the work done in this PR.  Check out the tests in the main branch. I only noticed it because i had to adjust those tests due to the API changes that resulted from the work in this PR.

Reply:
> It's not a build problem there is some other issue with the instrumentation tests outside of the work done in this PR. Check out the tests in the main branch.

All of the tests on the `main` branch are passing on CI and for me locally, see https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/10809/workflows/b38925bf-4997-4f7d-966c-3c3347d90466/jobs/42470.

> SanityUIRouteTest also fails

This one passes for me, the issue only seems to be with `selectAlternateRoute`.

I think we shouldn't move forward with ignoring this test, it reliably fails and has to highlight some kind of an issue with the PR.

Reply:
The tests fail for me locally on the main branch.

android.view.InflateException: Binary XML file line #7 in com.mapbox.navigation.instrumentation_tests:layout/activity_basic_navigation_view: Binary XML file line #7 in com.mapbox.navigation.instrumentation_tests:layout/activity_basic_navigation_view: Error inflating class com.mapbox.maps.MapView
Caused by: android.view.InflateException: Binary XML file line #7 in com.mapbox.navigation.instrumentation_tests:layout/activity_basic_navigation_view: Error inflating class com.mapbox.maps.MapView
Caused by: java.lang.reflect.InvocationTargetException


After further discussion perhaps this is an issue with running the tests on the emulator.

Reply:
Any updates here @cafesilencio?

Reply:
The main tests pass on my device so I"ve been working on the route selection test. It's challenging because two of the API calls are now asynchronous.